tests/dogtag/pytest-ansible/pytest/ca/bugzilla/test_bug_1787115_Copy_SKI_from_CSR_to_Certificate_Signed.py

```diff
@@ -0,0 +1,257 @@
+
+			:ExpectedResults:
+				1. CSR should be generate with SKI extension.
+				2. CSR should be submitted succeesfully.
+				3. Request should be approved succeesfully.
+				4. Signed Certificate should be exported succeesfully.
+				5. CSR and Signed Certificate SKI extension should identical.
+	"""
+	request_id = None
+	submitted_req = False
+	SKI_CSR = '/opt/pki/certdb/SKI.csr'
+	SKI_CRT = '/opt/pki/certdb/SKI.crt'
+	CA_SKID = '0xcba7aa1e3d2784153d47'  # random hexadecimal number starts with 0x
+	
+	command = ['echo {} > /opt/pki/certdb/passwd.txt'.format(constants.CLIENT_DATABASE_PASSWORD),
+	           'openssl rand -out {}/noise.bin 2048'.format(constants.NSSDB)]
+	for x in command:
+		output = ansible_module.shell(x)
+		print("Create noise file: %s", x)
+	
+	certutil_args = {'-d ': '{}'.format(constants.NSSDB), '-f': '{}/passwd.txt'.format(constants.NSSDB),
+	                 '-z': '{}/noise.bin'.format(constants.NSSDB), '-k': 'rsa', '-g': '2048', '-Z': 'SHA256',
+	                 '-s': 'CN=pki1.example.com,O=EXAMPLE',
+	                 '--keyUsage': 'critical,dataEncipherment,keyEncipherment,digitalSignature',
+	                 '--extKeyUsage': 'serverAuth', '-o': '{}.der'.format(SKI_CSR), '--extSKID': ''}
+	
+	log.info("Generating certificate request using certutil tool")
+	data = []
+	for key, value in certutil_args.items():
+		data.append(key)
+		data.append(value)
+		command = 'certutil' + ' ' + '-R' + " " + " ".join(data)
+	
+	log.info("Running Certutil command : %s", command)
+	csr_output = ansible_module.expect(command=command, responses={
+		"Enter value for the key identifier fields,enter to omit:": "{}".format(CA_SKID),
+		"Is this a critical extension [y/N]?": "y"})
+	for result in csr_output.values():
+		if result['rc'] == 0:
+			assert 'Generating key.  This may take a few moments...' in result['stdout']
+		else:
+			pytest.fail("Failed to Run Command {}".format(command))
+	
+	ansible_module.shell('openssl req -inform der -in {}.der -out {}'.format(SKI_CSR, SKI_CSR))
+	
+	sub_req = ansible_module.pki(cli='ca-cert-request-submit',
+	                             nssdb=constants.NSSDB,
+	                             port=constants.CA_HTTP_PORT,
+	                             dbpassword=constants.CLIENT_DATABASE_PASSWORD,
+	                             certnick='"{}"'.format(constants.CA_ADMIN_NICK),
+	                             extra_args='--profile {}  --csr-file {}'.format(profile, SKI_CSR))
+	for result in sub_req.values():
+		log.info("Running {}".format(result['cmd']))
+		if result['rc'] == 0:
+			assert "Submitted certificate request" in result['stdout']
+			assert "Request ID" in result['stdout']
+			assert "Request Status: pending" in result['stdout']
+			try:
+				request_id = re.search('Request ID: [\w]*',
+				                       result['stdout']).group().encode('utf-8')
+				request_id = request_id.split(":")[1].strip()
```

Comment:
Error: There seems to be some issue with your regex pattern. You may want to debug this 😄